apps/ecommerce/frontend/src/components/App/Field/ResourceField/ResourceField/ResourceField.tsx

```diff
@@ -8,26 +7,29 @@ import SortableResourceList from '../SortableResourceList/SortableResourceList';
 import useResourceValue from 'hooks/field/useResourceValue';
 import { FieldAppSDK } from '@contentful/app-sdk';
 import { getResourceProviderAndType } from 'helpers/resourceProviderUtils';
+import { styles } from './ResourceField.styles';
 
 const ResourceField = () => {
-  const { isMultiple, handleAddResource } = useContext(ResourceFieldContext);
+  const { isMultiple, handleAddResource, logoUrl } = useContext(ResourceFieldContext);
   const { value } = useResourceValue(isMultiple);
   const sdk = useSDK<FieldAppSDK>();
   const { linkType } = sdk.parameters.instance;
   const { resourceType } = getResourceProviderAndType(linkType);
-  const buttonText = isMultiple ? `Add ${resourceType}s` : `Add a ${resourceType}`;
+  const buttonText = isMultiple
+    ? `Add ${resourceType}s`
+    : value.length
+    ? `Edit ${resourceType}`
+    : `Add ${resourceType}`;
```

Comment:
Really like these changes you added in this file! A little nitpicky, but or this logic, I tend to avoid using nested ternaries for readability and complexity reasons, and fallback to a function with an if statement such as 

const getButtonText = () => {
  if (isMultiple) return `Add ${resourceType}s`;
  return value.length ? `Edit ${resourceType}` : `Add ${resourceType}`
}

or something along those lines!

Reply:
Good call!